bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -760,15 +783,25 @@ boolean refreshImageForZoom () {
 		}
 	} else {
 		if (deviceZoomLevel != currentDeviceZoom) {
-			ImageData data = getImageDataAtCurrentZoom();
-			destroy ();
-			ImageData resizedData = DPIUtil.autoScaleImageData(device, data, deviceZoomLevel, currentDeviceZoom);
-			init(resizedData);
-			init();
-			refreshed = true;
-			currentDeviceZoom = deviceZoomLevel;
+			if (this.dataAtBaseZoomFactor == null && memGC == null) {
+				// Cache data at base zoom before refresh.
+				this.dataAtBaseZoomFactor = getImageData(currentDeviceZoom);
+				this.dataBaseZoomFactor = this.currentDeviceZoom;
+			}
```

Comment:
Can you explain how this case can happen? I would expect the data for base zoom to always be initialized for an image, i.e., to be an invariant of an instance of this class.

Reply:
This happens when `Image::win32_new(...)` is used with a given handle, e.g. in `Display::getSystemImage`

Reply:
Okay, shouldn't/couldn't this functionality then be part of `getImageData(int)`?
And why is it okay that nothing done when `dataAtBaseZoomFactor == null` and `memGC != null`?

Reply:
Answer to the last question: we discussed that some weeks ago with the PaintExample. In this case we will have destructive modifications, If we would scale it. That ist why we excluded it for now.

Reply:
As discussed, I moved the caching into `getImageData(int)` and refactored the caller a bit